src/sentry/search/events/fields.py

```diff
@@ -1604,3 +1611,59 @@ def is_accessible(self, acl=None):
 
 
 FUNCTION_ALIAS_PATTERN = re.compile(r"^({}).*".format("|".join(list(FUNCTIONS.keys()))))
+
+
+class QueryFields(QueryBase):
+    """ Field logic for a snql query """
+
+    def resolve_select(self, selected_columns: List[str]) -> None:
+        for field in selected_columns:
+            if field.strip() == "":
+                continue
+            field = self.resolve_field(field)
+            if isinstance(field, SnqlColumn) and field not in self.columns:
+                self.columns.append(field)
+
+    def resolve_field(self, field) -> Union[SnqlColumn, SnqlFunction]:
+        if not isinstance(field, str):
+            raise InvalidSearchQuery("Field names must be strings")
+
+        match = is_function(field)
```

Comment:
A lot of this parsing code (functions, orderby etc.) is pretty close to what the SDK does in the `legacy` package. Maybe you could reuse some of those? I'm hoping ultimately to deprecate that entire package, but if leveraging it helps with the transition to SnQL then I think it would be fine.

Reply:
Hmmm would it be better to base our implementation on the `legacy` package if needed instead of using it here, that way we don't need to release/update the sdk if there's a need for us to change anything? (and so you can still deprecate it as needed)

Reply:
I think if the functionality in `legacy` is going to be useful generally, then I just won't deprecate it.

This opens up a larger question actually: When I wrote the SDK I had thought that a class like `QueryBuilder` wouldn't be necessary. That the SDK Query class would be all that was needed. It seems like that assumption wasn't really true though, since there is logic needed in an intermediary class like `QueryBuilder`. My question is: do you think that you will ever _not_ need the `QueryBuilder` class? Does the Query data structure from the SDK actually fit the needs you have?

Reply:
If we keep the `legacy` package around could we move it out of the sdk? 
Reasons:
- that package contains a number of hacks meant for the rollout. there is no clean way to fully transform a legacy query into a SnQL query mainly because the original legacy grammar is inherently ambiguous. 
- it contains features that are not complete and only meant to work in the specific cases we encountered during the rollout. That is not what you would like in a generic api. If that solves a specific product need, then it should be in the product.
- Portability: if we wrote a nodeJS or rust service that had to query snuba we would have to port the sdk. Then we would have to either port the legacy package as well, thus carrying over the technical debt it contain on other platforms or make the port partial, which will be a pain to use as the api provided would not be the same the python sdk provides.

Reply:
But also, to echo Evan, What's the gap between the sdk Query class and the QueryBuilder ? Is it just a matter of the syntax used by the user to provide the query? Then a translation layer somewhere is inevitable and it would be product specific (other areas would not need it). In such case is there a reason to have an intermediate representation between the parser and SnQL?

Reply:
So the QueryBuilder class exists more as a convenient way to maintain parameters/state while parsing functions/fields. The main motivation for it is the current `params` function parameter on the majority of the functions in `search/events/fields.py` and `search/events/functions.py`.

The class doesn't necessarily have to continue existing and could be replaced with a series of functions like we do today.